packages/web-components/fast-element/src/controller.ts

```diff
@@ -292,6 +320,68 @@ export class Controller extends PropertyChangeNotifier {
         return false;
     }
 
+    private finishInitialization() {
+        const element = this.element;
+        const boundObservables = this.boundObservables;
+
+        // If we have any observables that were bound, re-apply their values.
+        if (boundObservables !== null) {
+            const propertyNames = Object.keys(boundObservables);
+
+            for (let i = 0, ii = propertyNames.length; i < ii; ++i) {
+                const propertyName = propertyNames[i];
+                (element as any)[propertyName] = boundObservables[propertyName];
+            }
+
+            this.boundObservables = null;
+        }
+
+        const definition = this.definition;
+
+        if (this._template === null) {
+            if ((this.element as any).resolveTemplate) {
+                this._template = (this.element as any).resolveTemplate();
+            } else if (definition.template) {
+                this._template = definition.template || null;
+            }
+        }
+
+        if (this._template !== null) {
+            this.renderTemplate(this._template);
+        }
+
+        if (this._styles === null) {
+            if ((this.element as any).resolveStyles) {
+                this._styles = (this.element as any).resolveStyles();
+            } else if (definition.styles) {
+                this._styles = definition.styles || null;
+            }
+        }
+
+        if (this._styles !== null) {
+            this.addStyles(this._styles);
+        }
+
+        this.needsInitialization = false;
+    }
+
+    private renderTemplate(template: ElementViewTemplate | null | undefined) {
+        const element = this.element;
+        const host = getShadowRoot(element) || element;
+
+        if (this.view !== null) {
+            this.view.dispose();
+            (this as Mutable<this>).view = null;
+        } else if (!this.needsInitialization) {
+            host.innerHTML = "";
+        }
+
+        if (template) {
+            // we have a new template to render to the Light
```

Comment:
heh - the Light!

Reply:
Oops. 🤣  Old comment left around after I refactored. I pushed a commit that adds some better internal comments on what's going on thoughout.

Reply:
I mean, I've seen, written, and found worse comments @EisenbergEffect 😆